src/redux/reducers/messages.js

```diff
@@ -21,6 +22,19 @@ export default handleActions(
         unreadMessageIDs: [],
       };
     },
+    [actionTypes.MESSAGES_GET_USER_GEO_DATA](state, { payload }) {
+      return {
+        ...state,
+        userGeoData: {
+          /* Manual schema stiching to provide necessary fields that are required */
+          ...payload,
+          text: 'Your location',
+          link: `https://www.openstreetmap.org/#map=13/${payload.lat}/${
```

Comment:
We only require `lon, lat`, remove everything else from`userGeoData`

Reply:
But we need text and link to display above the card content. For eg. as `here is a map` should point to map link. That's why I did schema stitching there.

Reply:
Create the link and text in the place you are using it. Not in the reducer.

Reply:
Okay.

Reply:
Have pushed final commit @fragm3